### ISSUE TITLE
Inventory full cant join Server

### ISSUE THREAD

**Issue by Shinjix3** [StrangeLoopGames/EcoIssues#18174]
**Description:**
Not possible to join server after inventory is full (item slots)

**Reproduce:**
**Important** -> im an alpha baker! so im getting the steam hat and the meteor shard!
1. Fill all your slots including your hotbar with items (Dont fill it with claim papers)
2. Store the claim papers into any chest or storage
3. Just before a character level up, logout!
4.  Relog into the server

**Server Exception:**
```
--BEGIN DUMP--
Dump Time
09/10/2020 20:59:37

Exception
Exception: InvalidOperationException
Message:Changes could not be applied, the requested operation would result in an invalid inventory state: Nicht genug Platz im Inventar
Source:Eco.Gameplay

System.InvalidOperationException: Changes could not be applied, the requested operation would result in an invalid inventory state: Nicht genug Platz im Inventar
   at Eco.Gameplay.Items.InventoryChangeSet.ApplyInternal(Boolean throwOnInvalid)
   at Eco.Gameplay.Items.InventoryChangeSet.Apply()
   at Eco.Gameplay.Items.Inventory.AddItem(Type itemType, User user)
   at Eco.Gameplay.Players.User.FirstLogin()
   at Eco.Gameplay.Players.User.Login(Player player, INetClient client, Boolean allowDisconnectionAfterActionFailure, Boolean ignoreTestMode)
   at Eco.Gameplay.Players.Player..ctor(User user, Single viewDistance, INetClient client)
   at Eco.Plugins.Networking.Client.SpawnPlayer(Single viewDistance, String language)

--END DUMP--
```

**Client Exception:**
```
<size=60,00%>Exception: InvalidOperationException
Message:Changes could not be applied, the requested operation would result in an invalid inventory state: Nicht genug Platz im Inventar
Source:Eco.Gameplay

System.InvalidOperationException: Changes could not be applied, the requested operation would result in an invalid inventory state: Nicht genug Platz im Inventar
   at Eco.Gameplay.Items.InventoryChangeSet.ApplyInternal(Boolean throwOnInvalid)
   at Eco.Gameplay.Items.InventoryChangeSet.Apply()
   at Eco.Gameplay.Items.Inventory.AddItem(Type itemType, User user)
   at Eco.Gameplay.Players.User.FirstLogin()
   at Eco.Gameplay.Players.User.Login(Player player, INetClient client, Boolean allowDisconnectionAfterActionFailure, Boolean ignoreTestMode)
   at Eco.Gameplay.Players.Player..ctor(User user, Single viewDistance, INetClient client)
   at Eco.Plugins.Networking.Client.SpawnPlayer(Single viewDistance, String language)</size>
EcoEngine.Logging.LogManager:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogWarning(Object)
NetworkManager:OnClientDisconnected(String)
System.Action`1:Invoke(T)
Eco.Shared.Networking.NetworkClient:HandleDisconnect(String)
Eco.Shared.Networking.NetworkClient:HandleDisconnected(Object, Boolean)
System.Threading.WaitOrTimerCallback:Invoke(Object, Boolean)
Eco.Networking.ENet.ENetUdpClient:OnDisconnected(ENetConnection, Boolean)
Eco.Networking.ENet.ENetUdpPeer:HandleEvent()
Eco.Networking.ENet.ENetUdpPeer:Service(Int32)
<>c__DisplayClass59_0:<Initialize>b__0()
HighlightingSystem.OnEndOfFrame:Invoke()
System.Action`1:Invoke(T)
System.Threading.ExecutionContext:RunInternal(ExecutionContext, ContextCallback, Object, Boolean)
System.Threading.ExecutionContext:Run(ExecutionContext, ContextCallback, Object)
HighlightingSystem.OnEndOfFrame:Invoke()
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug/Debug.bindings.h Line: 35)

Unloading 7 Unused Serialized files (Serialized files now loaded: 0)

Unloading 30781 unused Assets to reduce memory usage. Loaded Objects now: 91348.
Total: 1211.387600 ms (FindLiveObjects: 37.817300 ms CreateObjectMapping: 31.191600 ms MarkObjects: 899.033900 ms  DeleteObjects: 243.342600 ms)
```

**My Opinion:**
I think that the alpha items will be given every time i login to my server. But currently there is no space in backpack or hotbar.

**Comment by Shinjix3** [comment]
Updated the client Exception.

**Comment by Shinjix3** [comment]
Okay, 
reproduce was updated.
Its not the alpha items. Its the claim papers just before level up!
I tested it now on a local server.

All slots were full in inventory and hotbar.
I relogged just before im getting a new level of my character.
So after level up i will get a new claim, thats why im not getting into my server.

**Comment by beastyn** [comment]
@Shinjix3 thank you for the report. You should not have claim papers just with levelup. So, it must be something else.
Can you also provide client log?

**Comment by Shinjix3** [comment]
Yes i discovered that.
The client log is attached in the main thread over there. There is nothing else in the log.

Yesterday i saw that i get alpha items a second time. I just restarted the server and logged in.
Had a Picture of that in discord:
https://discord.com/channels/254025510651297802/312991502697627658/754707041846689882



If its usefull, i copied the corrupted databse and .eco file and would send it to you.

**Comment by IcedForge** [comment]
Still needs to investigate why it is generating more items each time he logs on?
Possible db entry issue?

If this happens again, could you provide the server .eco and .db files ?
@Shinjix3

**Comment by Shinjix3** [comment]
I have the Game.db and .eco files saved.
You can download it here:
https://www.dropbox.com/s/vbuyily6rptriov/Survival.rar?dl=0

Hope it helps

**Comment by mirasrael** [comment]
This issue is duplicate of https://github.com/StrangeLoopGames/EcoIssues/issues/15545 which already fixed in 9.1.
